src/device/pf_block_writer.c

```diff
@@ -2400,3 +2413,464 @@ void pf_put_input_data(
+
+      /* 1 bytes padding if needed*/
+      if (net->lldp_peer_info.chassis_id_len % 2 != 0)
+      {
+         pf_put_byte(temp_u8, res_len, p_bytes, p_pos);
+      }
+
+      /* Line Delay */
+      pf_put_uint32(is_big_endian, net->lldp_peer_info.line_delay, res_len, p_bytes, p_pos);
+
+      /* PeerMAC Addr */
+      pf_put_mem(net->lldp_peer_info.mac_address.addr, sizeof(net->lldp_peer_info.mac_address.addr), res_len, p_bytes, p_pos);
+
+      /* Two bytes padding */
+      pf_put_uint16(is_big_endian, temp_u16, res_len, p_bytes, p_pos);
+
+      /* MAUType */
+      pf_put_uint16(is_big_endian, net->lldp_peer_info.phy_config.operational_mau_type, res_len, p_bytes, p_pos);
+
+      /* Two bytes padding */
+      pf_put_uint16(is_big_endian, temp_u16, res_len, p_bytes, p_pos);
+
+      /* Domain Boundary */
+      pf_put_uint32(is_big_endian, net->lldp_peer_info.domain_boundary, res_len, p_bytes, p_pos);
+
+      /* Multicast Boundary */
+      pf_put_uint32(is_big_endian, net->lldp_peer_info.multicast_boundary, res_len, p_bytes, p_pos);
+
+      /* LinkState.Port */
+      pf_put_byte(net->lldp_peer_info.link_state_port, res_len, p_bytes, p_pos);
+
+      /* LinkState.Link  */
+      /* TODO currently always set to up */
+      pf_put_byte(1, res_len, p_bytes, p_pos);
+
+      /* Two bytes padding */
+      pf_put_uint16(is_big_endian, temp_u16, res_len, p_bytes, p_pos);
+
+      /* MediaType (Decode what type it is and report it out per 
+       * PROFINET AL Protocol Table 717
+       */
+      switch (net->lldp_peer_info.phy_config.operational_mau_type)
+      {
+      case PNET_MAU_RADIO:
+         /* Radio */
+         pf_put_uint32(is_big_endian, PF_PD_MEDIATYPE_RADIO, res_len, p_bytes, p_pos);
+         break;
+      case PNET_MAU_COPPER_10BaseT:
+      case PNET_MAU_COPPER_100BaseTX_HALF_DUPLEX:
+      case PNET_MAU_COPPER_100BaseTX_FULL_DUPLEX:
+      case PNET_MAU_COPPER_1000BaseT_HALF_DUPLEX:
+      case PNET_MAU_COPPER_1000BaseT_FULL_DUPLEX:
+         /* Copper */
+         pf_put_uint32(is_big_endian, PF_PD_MEDIATYPE_COPPER, res_len, p_bytes, p_pos);
+         break;
+      case PNET_MAU_FIBER_100BaseFX_HALF_DUPLEX:
+      case PNET_MAU_FIBER_100BaseFX_FULL_DUPLEX:
+      case PNET_MAU_FIBER_1000BaseX_HALF_DUPLEX:
+      case PNET_MAU_FIBER_1000BaseX_FULL_DUPLEX:
+         /* Fiber */
+         pf_put_uint32(is_big_endian, PF_PD_MEDIATYPE_FIBER, res_len, p_bytes, p_pos);
+         break;
+      default:
+         /* Unknown */
+         pf_put_uint32(is_big_endian, PF_PD_MEDIATYPE_UNKNOWN, res_len, p_bytes, p_pos);
+         break;
+      }
+   }
+
+   /* Finally insert the block length into the block header */
+   block_len = *p_pos - (block_pos + 4);
+   block_pos += offsetof(pf_block_header_t, block_length); /* Point to correct place */
+   pf_put_uint16(is_big_endian, block_len, res_len, p_bytes, &block_pos);
+}
+
+void pf_put_pdport_statistics(
+    pnet_t                    *net,
```

Comment:
Could we send in interface_statistics instead of net?

Reply:
Fixed